File: back-end/appcentric-app/app/Models/Question.php
namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Question extends Model
{
    use HasFactory;

    protected $fillable = ['paper_id', 'question_text', 'marks', 'order'];

    protected $casts = [
        'marks' => 'integer',
        'order' => 'integer',
    ];

    public function paper()
    {
        return $this->belongsTo(Paper::class);
    }

    public function answers()
    {
        return $this->hasMany(Answer::class);
    }
}

File: appcentricafrica/lib/models/question.dart
import 'package:appcentricafrica/models/answer.dart';

class Question {
  final int id;
  final int paperId;
  final String questionText;
  final int marks;
  final int order;
  final List<Answer>? answers;

  Question({
    required this.id,
    required this.paperId,
    required this.questionText,
    required this.marks,
    required this.order,
    this.answers,
  });

  factory Question.fromJson(Map<String, dynamic> json) {
    return Question(
      id: json['id'],
      paperId: json['paper_id'],
      questionText: json['question_text'],
      marks: json['marks'],
      order: json['order'],
      answers: json['answers'] != null
          ? (json['answers'] as List).map((a) => Answer.fromJson(a)).toList()
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'paper_id': paperId,
      'question_text': questionText,
      'marks': marks,
      'order': order,
      'answers': answers?.map((a) => a.toJson()).toList(),
    };
  }
}